homeassistant/components/lastfm/sensor.py

```diff
@@ -35,95 +35,52 @@
 )
 
 
+def format_track(track: Track) -> str:
+    """Format the track."""
+    return f"{track.artist} - {track.title}"
+
+
 def setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Last.fm sensor platform."""
-    api_key = config[CONF_API_KEY]
-    users = config[CONF_USERS]
-
-    lastfm_api = lastfm.LastFMNetwork(api_key=api_key)
-
-    entities = []
-    for username in users:
-        try:
-            lastfm_api.get_user(username).get_image()
-            entities.append(LastfmSensor(username, lastfm_api))
-        except WSError as error:
-            _LOGGER.error(error)
-            return
-
-    add_entities(entities, True)
+    coordinator = LastFmUpdateCoordinator(hass, config)
+    add_entities((LastFmSensor(coordinator, user) for user in config[CONF_USERS]), True)
```

Comment:
```suggestion
    coordinator = LastFmUpdateCoordinator(hass, config)
    try:
        coordinator.async_resfresh()
    except ...:
        raise PlatformNotReady
    add_entities((LastFmSensor(coordinator, user) for user in config[CONF_USERS]))
```
Something like this to refresh coordinator data before adding entities and possibly raise `PlatformNotReady` if data not available.

Reply:
Only thing is that the setup_platform method is sync and not async. And I was planning to remove the whole platform setup in a later PR (#91601)

Reply:
You cannot remove it for at least a couple of releases (deprecation period).
It's ok to make it async now.